pywbem/_logging.py

```diff
@@ -117,279 +167,183 @@
-                                        log_filename)
-
-    @classmethod
-    def get_logger_info(cls, logger_name):
-        """
-        Get information about a logger by name.
-
-        Parameters:
-
-          logger_name (:term:`string`):
-            The pywbem logger name (ex. ``pywbem.ops``)
-
-        Returns:
-
-          Tuple with the following information in the tuple:
-          (log_detail_level, log_level, log_dest, log_filename)
-          or 'None' if the logger has not been defined to PywbemLoggers
-        """
-        return cls.loggers.get(logger_name, None)
-
-    @classmethod
-    def _parse_log_specs(cls, log_spec_str):
-        """
-        Parse a complete cmd line log specification that is in the
-        format defined in :meth:`~pywbem.PywbemLogger.create_logger`
-
-        Parameters:
-
-          log_spec_str (:term:`string`): The log spec string.
-
-        Returns:
-
-          Dictionary containing the parsed information where each entry is
-          log_comp (detail_level, dest)
-
-        Raises:
-
-          ValueError: The parsing of the input string failed.
-        """
-        log_specs_dict = dict()
-        log_specs = log_spec_str.split(',')
-        for log_spec in log_specs:
-            try:
-                log_component, log_data = log_spec.split("=", 1)
-            except ValueError:
-                log_component = log_spec
-                log_data = ''
-
-            log_values = log_data.split(":")
-
-            if not log_component:
-                raise ValueError("Log component name required in %s" % log_spec)
-
-            # cvt empty strings to None
-            log_values = [None if x == '' else x for x in log_values]
-            # expand to full size if not all values supplied
-            while len(log_values) < 2:
-                log_values.append(None)
-
-            if len(log_values) > 2:
-                raise ValueError("Invalid log detail. %s too many "
-                                 "components."
-                                 % log_spec)
-            log_specs_dict[log_component] = tuple(log_values)
-        return log_specs_dict
+            raise ValueError('Invalid log_component %s' % log_name)
+
+        if log_suffix:
+            logger_name = '%s.%s' % (logger_name, log_suffix)
+
+        # create named logger. We allow only a single handler for
+        # any logger so must remove any existing handler before adding
+        #
+        if handler:
+            handler.setFormatter(logging.Formatter(format_string))
+            logger = logging.getLogger(logger_name)
+            for hdlr in logger.handlers:
+                logger.removeHandler(hdlr)
+            logger.addHandler(handler)
+            logger.setLevel(logging.DEBUG)
 
 
 def get_logger(logger_name):
     """
     **Experimental:** *New in pywbem 0.11 as experimental.*
 
+    Return a :class:`~py:logging.Logger` object with the specified name.
```

Comment:
Shouldn't we support the "component name" on this function (or however the term will be that we use), instead of the Python logger name?

Reply:
I think I want to hide this whole function since it is only used internally and within the recorder.  Will do that

Reply:
WILL DO.